web_external/views/body/DataPanel.js

```diff
@@ -507,6 +514,33 @@ export default Panel.extend({
         events.trigger('m:toggle-bounds-label');
     },
 
+    applyNameFilter(e) {
+        var keyword = e.target.value;
+        if (this.nameFilterKeyword === keyword) {
+            return;
+        }
+        this.nameFilterKeyword = keyword;
+        if (keyword) {
+            this.filters.name = this.collection.models
+                .reduce((ids, dataset) => {
+                    var match = false;
+                    try {
+                        var regex = new RegExp(keyword, 'i');
+                        match = !!regex.exec(dataset.get('name'));
+                    } catch (ex) {
+                        match = dataset.get('name').toLocaleLowerCase().indexOf(keyword) !== -1;
```

Comment:
I recommend moving this below of the catch as `match |= dataset.get('name').toLocaleLowerCase().indexOf(keyword) !== -1;`.  Otherwise, if I have a dataset called "foo.bar" and search for "foo.b", it won't be found (because as a regex that doesn't match), which would be surprising to a user.

Reply:
To be clearer:
```
                    var match = false;
                    try {
                        var regex = new RegExp(keyword, 'i');
                        match = !!regex.exec(dataset.get('name'));
                    } catch (ex) {}
                    match |= dataset.get('name').toLocaleLowerCase().indexOf(keyword) !== -1;
                    if (match) {
                        ids.push(dataset.get('_id'));
                    }
                    return ids;
```